daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/processors/parsers/Parser.scala

```diff
@@ -110,21 +109,66 @@ trait Parser
        * ProcessingError it throws a new SDE.
        * */
       case pe: ParseError => pstate.setFailed(pe)
+    } finally {
+      pstate.resetFormatInfoCaches()
     }
     if (pstate.dataProc.isDefined) pstate.dataProc.get.after(pstate, this)
     pstate.setMaybeProcessor(savedParser)
   }
 }
 
-// Deprecated and to be phased out. Use the trait Parser instead.
-abstract class ParserObject(override val context: RuntimeData)
+/**
+ * A PrimParser is a parser that contains no child parsers.
+ * Combinators are NOT PrimParser
+ *
+ * This trait is preferred over PrimParserObject.
+ */
+trait PrimParser
+  extends PrimProcessor
+  with Parser
+
+trait PrimParserNoData
+  extends PrimProcessorNoData
+  with Parser
+
+trait TextPrimParser
+  extends PrimParser
+  with TextProcessor
+
+// Deprecated and to be phased out. Use the trait PrimParser instead.
```

Comment:
Why deprectated - because it is too easy to forget you have to define runtimeDependencies. 

All these objects do for you is define val context, and provide a default runtimeDependencies that is empty. They're probably not worth having. It's not a huge change because most of the time only a base class extends these base primitive objects directly.

Reply:
Is it worth removing them as part of this? Looks like there's only a dozen or so left